tools/clang/lib/SPIRV/SpirvModule.cpp

```diff
@@ -29,51 +29,129 @@ bool SpirvModule::invokeVisitor(Visitor *visitor, bool reverseOrder) {
   if (!visitor->visit(this, Visitor::Phase::Init))
     return false;
 
-  for (auto *cap : capabilities)
-    if (!cap->invokeVisitor(visitor))
+  if (reverseOrder) {
+    // Reverse order of a SPIR-V module.
+
+    // Our transformations do not cross function bounaries, therefore the order
+    // of visiting functions is not important.
+    for (auto fn : functions) {
```

Comment:
`for (auto iter = functions.rbegin(); iter != functions.rend(); ++iter) {` ??

Reply:
done